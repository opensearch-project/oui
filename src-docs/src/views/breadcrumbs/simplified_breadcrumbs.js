/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import { OuiSimplifiedBreadcrumbs } from '../../../../src/components';

import { DisplayBreadCrumbsToggles } from './display_breadcrumbs_toggles';

export default () => {
  const breadcrumbs = [
    {
      text: 'Universe',
      href: '#',
    },
    {
      text: 'Observable Universe',
      href: '#',
    },
    {
      text: 'Piscesâ€“Cetus Supercluster Complex',
      href: '#',
    },
    {
      text: 'Laniakea Supercluster',
      href: '#',
    },
    {
      text: 'Virgo Cluster',
      href: '#',
    },
    {
      text: 'Local Group',
      href: '#',
    },
    {
      text: 'Local Bubble',
      href: '#',
    },
    {
      text: 'Local Interstellar Cloud',
      href: '#',
    },
    {
      text: 'Milky Way Galaxy',
      href: '#',
    },
    {
      text: 'Orion Arm',
      href: '#',
    },
    {
      text: 'Solar System',
      href: '#',
    },
    {
      text: 'Geospace',
      href: '#',
    },
    {
      text: 'Earth',
      href: '#',
    },
  ];

  return (
    <DisplayBreadCrumbsToggles
      canHideLastBreadCrumb
      canHideTrailingSeparator
      canDisableTrailingLink>
      <OuiSimplifiedBreadcrumbs breadcrumbs={breadcrumbs} max={null} />
    </DisplayBreadCrumbsToggles>
  );
};
