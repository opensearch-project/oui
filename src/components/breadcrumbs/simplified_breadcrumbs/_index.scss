/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

@import 'simplified_breadcrumbs';
