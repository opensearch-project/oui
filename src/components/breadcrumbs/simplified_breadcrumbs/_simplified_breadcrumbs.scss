/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

.ouiSimplifiedBreadcrumbs {
  @include ouiFontSizeS;
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  gap: $ouiSizeS; // sass-lint:disable-line no-misspelled-properties
  min-width: 0; // Ensure it shrinks if the window is narrow
  margin-left: 0;

  .ouiLink.euiLink--text:focus {
    animation: none;
    background: none;
  }

  &--truncate {
    white-space: nowrap;
    flex-wrap: nowrap;

    .ouiSimplifiedBreadcrumb:not(.ouiSimplifiedBreadcrumb--collapsed) {
      width: 100%;
      max-width: fit-content;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
}

.ouiSimplifiedBreadcrumb {
  display: block;

  &.ouiLink {
    line-height: inherit;
    font-weight: inherit;

    &[class*='ouiLink--']:focus {
      // .ouiLink uses a rule with !important
      background-color: unset !important; // sass-lint:disable-line no-important
      animation: none !important; // sass-lint:disable-line no-important
    }
  }

  &--collapsed {
    flex-shrink: 0;
    color: $ouiBreadcrumbCollapsedLink !important; // sass-lint:disable-line no-important
    vertical-align: top !important; // sass-lint:disable-line no-important

    &:hover {
      color: $ouiBreadCrumbHoverColor !important; // sass-lint:disable-line no-important
    }
  }

  &--truncate {
    @include ouiTextTruncate;
    max-width: 100%;
    text-align: center;
    vertical-align: top; // overflow hidden causes misalignment of links and slashes, this fixes that
  }
}

.ouiBreadcrumbSeparator {
  flex-shrink: 0;
  width: 8px; // The width of the SVG's bounding box
  color: $ouiColorDisabledText;
  height: $ouiFontSizeXS; // One size smaller the font size used by .ouiSimplifiedBreadcrumbs
}
