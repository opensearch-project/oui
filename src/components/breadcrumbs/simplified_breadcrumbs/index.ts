/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export {
  OuiSimplifiedBreadcrumbs,
  OuiSimplifiedBreadcrumbsProps,
} from './simplified_breadcrumbs';
