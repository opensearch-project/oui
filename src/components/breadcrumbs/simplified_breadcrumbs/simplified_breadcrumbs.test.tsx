/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test';
import { OuiSimplifiedBreadcrumbs } from './simplified_breadcrumbs';

const breadcrumbs = [
  {
    text: 'Animals',
    href: '#',
    onClick: (e: React.MouseEvent) => {
      e.preventDefault();
      console.log('You clicked Animals');
    },
    'data-test-subj': 'breadcrumbsAnimals',
    className: 'customClass',
  },
  {
    text: 'Metazoans',
  },
  {
    text: 'Chordates',
  },
  {
    text:
      'Nebulosa subspecies is also a real mouthful, especially for creatures without mouths',
    truncate: true,
  },
  {
    text: 'Tetrapods',
  },
  {
    text: 'Reptiles',
    onClick: (e: React.MouseEvent) => {
      e.preventDefault();
      console.log('You clicked Reptiles');
    },
  },
  {
    text: 'Boa constrictor',
    href: '#',
    truncate: true,
  },
  {
    text: 'Edit',
  },
];

describe('OuiSimplifiedBreadcrumbs', () => {
  test('is rendered', () => {
    const component = render(
      <OuiSimplifiedBreadcrumbs {...requiredProps} breadcrumbs={breadcrumbs} />
    );

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    describe('responsive', () => {
      test('is rendered', () => {
        const component = render(
          <OuiSimplifiedBreadcrumbs breadcrumbs={breadcrumbs} responsive />
        );
        expect(component).toMatchSnapshot();
      });

      test('is rendered as false', () => {
        const component = render(
          <OuiSimplifiedBreadcrumbs
            breadcrumbs={breadcrumbs}
            responsive={false}
          />
        );
        expect(component).toMatchSnapshot();
      });

      test('is rendered with custom breakpoints', () => {
        const component = render(
          <OuiSimplifiedBreadcrumbs
            breadcrumbs={breadcrumbs}
            responsive={{ xs: 1, s: 1, m: 1, l: 1, xl: 1 }}
          />
        );
        expect(component).toMatchSnapshot();
      });
    });

    describe('truncate as false', () => {
      test('is rendered', () => {
        const component = render(
          <OuiSimplifiedBreadcrumbs
            breadcrumbs={breadcrumbs}
            truncate={false}
          />
        );
        expect(component).toMatchSnapshot();
      });
    });

    describe('max', () => {
      test('renders 1 item', () => {
        const component = render(
          <OuiSimplifiedBreadcrumbs breadcrumbs={breadcrumbs} max={1} />
        );
        expect(component).toMatchSnapshot();
      });

      test('renders all items with null', () => {
        const component = render(
          <OuiSimplifiedBreadcrumbs breadcrumbs={breadcrumbs} max={null} />
        );
        expect(component).toMatchSnapshot();
      });

      test("doesn't break when max exceeds the number of breadcrumbs", () => {
        const component = render(
          <OuiSimplifiedBreadcrumbs breadcrumbs={breadcrumbs} max={20} />
        );
        expect(component).toMatchSnapshot();
      });
    });
  });
});
