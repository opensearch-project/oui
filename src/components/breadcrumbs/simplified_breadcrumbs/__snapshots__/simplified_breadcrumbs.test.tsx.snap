// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiSimplifiedBreadcrumbs is rendered 1`] = `
<nav
  aria-label="aria-label"
  class="ouiSimplifiedBreadcrumbs testClass1 testClass2 ouiSimplifiedBreadcrumbs--truncate"
  data-test-subj="test subject string"
>
  <div
    class="ouiSimplifiedBreadcrumbWall"
  >
    <a
      class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb customClass"
      data-test-subj="breadcrumbsAnimals"
      href="#"
      rel="noreferrer"
    >
      Animals
    </a>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Metazoans
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <button
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb"
    type="button"
  >
    Reptiles
  </button>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <a
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--truncate"
    href="#"
    rel="noreferrer"
  >
    Boa constrictor
  </a>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;

exports[`OuiSimplifiedBreadcrumbs props max doesn't break when max exceeds the number of breadcrumbs 1`] = `
<nav
  aria-label="breadcrumb"
  class="ouiSimplifiedBreadcrumbs ouiSimplifiedBreadcrumbs--truncate"
>
  <div
    class="ouiSimplifiedBreadcrumbWall"
  >
    <a
      class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb customClass"
      data-test-subj="breadcrumbsAnimals"
      href="#"
      rel="noreferrer"
    >
      Animals
    </a>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Metazoans
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Chordates
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <button
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb"
    type="button"
  >
    Reptiles
  </button>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <a
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--truncate"
    href="#"
    rel="noreferrer"
  >
    Boa constrictor
  </a>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;

exports[`OuiSimplifiedBreadcrumbs props max renders 1 item 1`] = `
<nav
  aria-label="breadcrumb"
  class="ouiSimplifiedBreadcrumbs ouiSimplifiedBreadcrumbs--truncate"
>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;

exports[`OuiSimplifiedBreadcrumbs props max renders all items with null 1`] = `
<nav
  aria-label="breadcrumb"
  class="ouiSimplifiedBreadcrumbs ouiSimplifiedBreadcrumbs--truncate"
>
  <div
    class="ouiSimplifiedBreadcrumbWall"
  >
    <a
      class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb customClass"
      data-test-subj="breadcrumbsAnimals"
      href="#"
      rel="noreferrer"
    >
      Animals
    </a>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Metazoans
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Chordates
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <button
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb"
    type="button"
  >
    Reptiles
  </button>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <a
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--truncate"
    href="#"
    rel="noreferrer"
  >
    Boa constrictor
  </a>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;

exports[`OuiSimplifiedBreadcrumbs props responsive is rendered 1`] = `
<nav
  aria-label="breadcrumb"
  class="ouiSimplifiedBreadcrumbs ouiSimplifiedBreadcrumbs--truncate"
>
  <div
    class="ouiSimplifiedBreadcrumbWall"
  >
    <a
      class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb customClass"
      data-test-subj="breadcrumbsAnimals"
      href="#"
      rel="noreferrer"
    >
      Animals
    </a>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Metazoans
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <button
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb"
    type="button"
  >
    Reptiles
  </button>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <a
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--truncate"
    href="#"
    rel="noreferrer"
  >
    Boa constrictor
  </a>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;

exports[`OuiSimplifiedBreadcrumbs props responsive is rendered as false 1`] = `
<nav
  aria-label="breadcrumb"
  class="ouiSimplifiedBreadcrumbs ouiSimplifiedBreadcrumbs--truncate"
>
  <div
    class="ouiSimplifiedBreadcrumbWall"
  >
    <a
      class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb customClass"
      data-test-subj="breadcrumbsAnimals"
      href="#"
      rel="noreferrer"
    >
      Animals
    </a>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Metazoans
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <button
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb"
    type="button"
  >
    Reptiles
  </button>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <a
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--truncate"
    href="#"
    rel="noreferrer"
  >
    Boa constrictor
  </a>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;

exports[`OuiSimplifiedBreadcrumbs props responsive is rendered with custom breakpoints 1`] = `
<nav
  aria-label="breadcrumb"
  class="ouiSimplifiedBreadcrumbs ouiSimplifiedBreadcrumbs--truncate"
>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;

exports[`OuiSimplifiedBreadcrumbs props truncate as false is rendered 1`] = `
<nav
  aria-label="breadcrumb"
  class="ouiSimplifiedBreadcrumbs"
>
  <div
    class="ouiSimplifiedBreadcrumbWall"
  >
    <a
      class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb customClass"
      data-test-subj="breadcrumbsAnimals"
      href="#"
      rel="noreferrer"
    >
      Animals
    </a>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="false"
    class="ouiSimplifiedBreadcrumb"
  >
    Metazoans
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <div
    class="ouiPopover ouiPopover--anchorDownCenter ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--collapsed"
  >
    <div
      class="ouiPopover__anchor"
    >
      <button
        aria-label="Show collapsed breadcrumbs"
        class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb__collapsedLink"
        title="Show collapsed breadcrumbs"
        type="button"
      >
        … 
        <span
          data-ouiicon-type="arrowDown"
        />
      </button>
    </div>
  </div>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <button
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb"
    type="button"
  >
    Reptiles
  </button>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <a
    class="ouiLink ouiLink--primary ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--truncate"
    href="#"
    rel="noreferrer"
  >
    Boa constrictor
  </a>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-current="page"
    class="ouiSimplifiedBreadcrumb ouiSimplifiedBreadcrumb--last"
  >
    Edit
  </span>
  <svg
    class="ouiBreadcrumbSeparator"
    preserveAspectRatio="none"
    viewBox="0 0 8 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 12H0L6 0h2z"
      fill="currentColor"
    />
  </svg>
</nav>
`;
